tests/qeqiskit/optimizer/optimizer_test.py

```diff
@@ -46,3 +62,49 @@ def test_optimizer_succeeds_on_cost_function_without_gradient(
         assert "opt_value" in results
         assert "opt_params" in results
         assert "history" in results
+
+
+class TestQiskitSinusoidalOptimizerTests(OptimizerTests):
+    """Some optimizers assume that the cost function can be expressed as a sum of
+    products of sines and cosines (Eq. 12 of arXiv:1903.12166), a functional form which
+    reflects the cost function of many variational quantum algorithms. Such optimizers
+    may fail to find the minimum of cost functions that are not of this form.
+
+    This class removes the tests from the base class that check whether the optimizer
+    finds the minimum of such non-sinusoidal cost functions, and replaces them with a
+    test on a sinuisoidal cost function."""
+
+    @pytest.fixture
+    def sum_cos_x(self):
+        """Create a sinuisoidal function whose minimum is at the origin with a value of
+        zero."""
+        return lambda x: -np.sum(np.cos(x)) + x.shape[0]
```

Comment:
Why using `lambda` here instead of just doing:
```
    @pytest.fixture
     def sum_cos_x(x):
         """Create a sinuisoidal function whose minimum is at the origin with a value of
         zero."""
         return -np.sum(np.cos(x)) + x.shape[0]
```

Reply:
lambdas tend to be (sooner or later) problematic with `mypy` (and `flake8` as well AFAIR)

Reply:
I was following the pattern from the [base class](https://github.com/zapatacomputing/z-quantum-core/blob/b1e68b38f7d26ad010f1eafb8d42fde29c0a7752/src/python/zquantum/core/interfaces/optimizer_test.py#L44-L53). I think the reason for the lambda is that pytest will call the fixture before passing it to a test case.

Reply:
Ok, sounds good!
I just remember removing a couple of lambdas here and there while doing the cleaning recently.